From 635725d1d46ea2065aa11f2270d6551fd169ec36 Mon Sep 17 00:00:00 2001
From: Saj Goonatilleke <sg@redu.cx>
Date: Mon, 9 Jul 2012 18:24:50 +1000
Subject: [PATCH 1005/1010] Make the 'bind' config parameter optional

With CONFIG SET BIND, it makes sense to allow a server to start without
listening on a TCP socket.  (We can connect to a Unix-domain socket to
reconfigure it at runtime.)  Since we might start listening on a TCP
port later, setting 'port' to 0 doesn't make sense, so remove that
behaviour while I'm at it.

In summary, your options are:

    1) #bind 127.0.0.1

       A commented-out 'bind' parameter is as good as no 'bind'
       parameter.  Fall-back on IPADDR_ANY.

    2) bind 127.0.0.1

       Bind to a single address:  127.0.0.1.

    3) bind ""

       Don't bind to a TCP socket.  This is new behaviour.  To use this
       option, you must supply a valid 'unixsocket'.
---
 src/redis.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/redis.c b/src/redis.c
index 164db46..60b5f01 100644
--- a/src/redis.c
+++ b/src/redis.c
@@ -1252,7 +1252,7 @@ void initServer() {
     server.el = aeCreateEventLoop(server.maxclients+1024);
     server.db = zmalloc(sizeof(redisDb)*server.dbnum);
 
-    if (server.port != 0) {
+    if (!server.bindaddr || (server.bindaddr && strlen(server.bindaddr) > 0)) {
         server.ipfd = anetTcpServer(server.neterr,server.port,server.bindaddr);
         if (server.ipfd == ANET_ERR) {
             redisLog(REDIS_WARNING, "Opening port %d: %s",
-- 
1.7.11.1

